pkg/api/types.go

```diff
@@ -2049,6 +2063,23 @@ type ServiceProxyOptions struct {
 	Path string
 }
 
+// OwnerReference contains enough information to let you identify an owning
+// object. Currently, an owning object must be in the same namespace, so there
+// is no namespace field.
+type OwnerReference struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion,omitempty"`
```

Comment:
The garbage collector *must* know how resources are located, otherwise it will be massively inefficient and unable to properly propagate deletions.

Also, as Chao says, since not all resources may exist in all versions, and one could even hypothetically have v1.Foo and v3.Foo be actually different types, you actually *do* need a version specified.

Additionally, we always specify this information this way in our API. I think we should continue to be consistent. I think this should stay as written.